homeassistant/components/samsungtv/bridge.py

```diff
@@ -534,16 +534,21 @@ async def _async_get_remote_under_lock(self) -> SamsungTVWSAsyncRemote | None:
             )
             try:
                 await self._remote.start_listening(self._remote_event)
-            except ConnectionClosedError as err:
-                # This is only happening when the auth was switched to DENY
-                # A removed auth will lead to socket timeout because waiting
-                # for auth popup is just an open socket
+            except UnauthorizedError as err:
                 LOGGER.info(
                     "Failed to get remote for %s, re-authentication required: %s",
                     self.host,
                     err.__repr__(),
                 )
                 self._notify_reauth_callback()
+                self._remote = None
+            except ConnectionClosedError as err:
+                LOGGER.info(
+                    "Failed to get remote for %s: %s",
+                    self.host,
+                    err.__repr__(),
```

Comment:
Why do we need to use the magic method? Isn't the str magic method used implicitly enough?

Reply:
Will see if we can swap it out tomorrow.

Reply:
https://github.com/home-assistant/core/pull/68817